source/validate.h

```diff
@@ -43,6 +45,26 @@
 #include "spirv_definition.h"
 #include "table.h"
 
+// clang-format off
+#define _printf_  printf
+#define MSG(msg,...) do {                       \
+    _printf_(__FILE__":%d: " msg "\n",          \
+             __LINE__, ##__VA_ARGS__);          \
+  } while (0)
+
+#ifdef DEBUG
+#define MSG_DBG      MSG
+#else
+#define MSG_DBG(...) do { /* nothing */ } while (0)
+#endif
+
+
+#define SHOW(exp)   do { MSG("%s  %ld",   #exp, (long)exp);   } while (0)
+#define SHOW_P(ptr) do { MSG("%s  %#zx",  #ptr, (size_t)ptr); } while (0)
+#define SHOW_F(exp) do { MSG("%s  %g",    #exp, (double)exp); } while (0)
+#define SHOW_S(exp) do { MSG("%s  %s",    #exp, exp);         } while (0)
+// clang-format on
```

Comment:
Do you think it would be detrimental to leave these in the code? I found these to be useful for debugging in many situations. I would like to avoid adding these every time i want to make changes to the project.